app/internal/features.py

```diff
@@ -0,0 +1,146 @@
+from fastapi import Depends
+from typing import List
+
+from app.database.models import UserFeature, Feature
+from app.dependencies import get_db, SessionLocal
+from app.features.index import features
+from app.internal.utils import create_model, get_current_user
+
+
+def create_features_at_startup(session: SessionLocal) -> bool:
+    for feat in features:
+        if not is_feature_exists_in_db(feature=feat, session=session):
+            create_feature(**feat, db=session)
+
+    return True
+
+
+def is_association_exists_in_db(form: dict, session: SessionLocal) -> bool:
+    db_association = session.query(UserFeature).filter_by(
+        feature_id=form['feature_id'],
+        user_id=form['user_id']
+    ).first()
+
+    return db_association is not None
+
+
+def delete_feature(
+    feature: Feature, session: SessionLocal = Depends(get_db)
+) -> None:
+    session.query(UserFeature).filter_by(feature_id=feature.id).delete()
+    session.query(Feature).filter_by(id=feature.id).delete()
+    session.commit()
+
+
+def is_feature_exists_in_db(feature: dict, session: SessionLocal) -> bool:
+    db_feature = session.query(Feature).filter(
+        (Feature.name == feature['name']) |
+        (Feature.route == feature['route'])).first()
+
+    if db_feature is None:
+        return False
+
+    # Update if found
+    update_feature(
+        feature=db_feature, new_feature_obj=feature, session=session)
```

Comment:
Why? We shouldn't have side-effects on `is_feature_exists`

Reply:
If the developer changed the index file with all the features info it's means he wants to update some thing this is an ease way to do so.

i wrote it in the documentation of the index file.

Reply:
If the developer calls is_feature_exists he or she definitely doesn't want to change the state of the database.

Reply:
Ok, I'll change it.